File: tcp_client.py
import socket

def start_tcp_client():
    try:
        client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
        
        client_socket.connect(('localhost', 8080))

        message = "Hello from Sameer"
        client_socket.sendall(message.encode())

        data = client_socket.recv(1024)
        print("Received from server:", data.decode())
    
    except Exception as e:
        print(f"Error: {e}")
    finally:
        client_socket.close()

if __name__ == "__main__":
    start_tcp_client()

File: tcp_client.cpp
#include <iostream>
#include <cstring>
#include <sys/socket.h>
#include <arpa/inet.h>
#include <unistd.h>
using namespace std;

#define PORT 8080
#define BUFFER_SIZE 1024

int main() {
    int sock = 0;
    struct sockaddr_in serv_addr;
    char buffer[BUFFER_SIZE] = {0};

    // Creating socket
    if ((sock = socket(AF_INET, SOCK_STREAM, 0)) < 0) {
        perror("Socket creation error");
        return -1;
    }

    serv_addr.sin_family = AF_INET;
    serv_addr.sin_port = htons(PORT);

    // Convert IPv4 and IPv6 addresses from text to binary form
    if (inet_pton(AF_INET, "127.0.0.1", &serv_addr.sin_addr) <= 0) {
        perror("Invalid address/ Address not supported");
        return -1;
    }

    // Connect to server
    if (connect(sock, (struct sockaddr*)&serv_addr, sizeof(serv_addr)) < 0) {
        perror("Connection failed");
        return -1;
    }

    // Send message to server
    const char* message = "Hello from client!";
    send(sock, message, strlen(message), 0);
    cout << "Message sent to server" << endl;

    // Receive response from server
    read(sock, buffer, BUFFER_SIZE);
    cout << "Message from server: " << buffer << endl;

    // Close the socket
    close(sock);

    return 0;
}
